packages/protocol/src/utils/StorageNodeRegistry.ts

```diff
@@ -1,20 +1,16 @@
-import { Contract, providers } from 'ethers'
-import { ConnectionInfo } from 'ethers/lib/utils'
-
 import * as storageNodeRegistryConfig from '../../contracts/NodeRegistry.json'
-
-const { JsonRpcProvider } = providers
+import { initContract, SmartContractConfig } from './SmartContractUtil'
 
 // storageNodeAddress => HTTP address
-export type StorageNodeInfo = {
+export type StorageNodeRecord = {
     address: string
     url: string
 }
 
 export class StorageNodeRegistry {
```

Comment:
Maybe we could move this class to `Broker` package as storage functionality is nowadays considered to be an implementation detail of a broker node, and not protocol level logic.

(At some point we should update the https://github.com/streamr-dev/streamr-specs/blob/master/PROTOCOL.md#storagenodesrequest spec and officially deprecate `StorageNodesRequest` and `StorageNodesRequest`)

There is already a class `network-monorepo/packages/broker/src/StorageNodeRegistry.ts`. This functionality could be merged to that file.

Reply:
In the decentralised version I am working on, the StorageNodeRegistry is residing (like the StreamRegistry) in the client and the one in the broker is being replaced by just calling the client. So this StorageNodeRecord is needed in the client for CRUD operations on storagenodes or adding streams to storagenodes. So it could be moved to the client, and then the broker uses the definition from the client instead of the protocol..